lib/services/page-service.ts

```diff
@@ -38,47 +42,24 @@ export class PageService implements IPageService {
         });
     }
 
-    private ensurePageTemplate(pageName: string, flavor: string): Promise<string> {
-        const templatesDir = util.path.join(__dirname, "../templates");
-        const pagePath = util.path.join(templatesDir, pageName);
-
-        return new Promise((resolve, reject) => {
-            util.fs.ensureDir(templatesDir)
-                .then(() => {
-                    return util.pageExists(templatesDir, pageName);
-                })
-                .then((pageExists: boolean) => {
-                    if (!pageExists) {
-                        return this.clonePageTemplate(pageName, flavor, templatesDir);
-                    }
-                })
-                .then(() => {
-                    resolve(pagePath);
-                })
-                .catch((ensurePageError: any) => {
-                    reject(ensurePageError);
-                });
-        });
-    }
-
-    private clonePageTemplate(pageName: string, flavor: string, templatesDir: string): Promise<string> {
+    private clonePageTemplate(pageName: string, flavor: string, templatesDirectory: string): Promise<string> {
         const command = "git";
-        const commandArgs: Array<any> = ["clone"];
+        const commandArguments: Array<any> = ["clone"];
 
         return new Promise((resolve, reject) => {
             util.getPageTemplatesBaseUrl(flavor)
                 .then((baseUrl: string) => {
                     baseUrl = baseUrl + ".git";
-                    commandArgs.push(baseUrl);
-                    commandArgs.push(templatesDir);
+                    commandArguments.push(baseUrl);
+                    commandArguments.push(templatesDirectory);
 
-                    const process = util.childProcess.spawn(command, commandArgs);
+                    const process = util.childProcess.spawn(command, commandArguments);
                     process.on("close", (code) => {
                         if (code !== 0) {
                             return Promise.reject(new Error(`child process exited with code ${code}`));
                         }
 
-                        const pagePath = util.path.join(templatesDir, pageName);
+                        const pagePath = util.path.join(templatesDirectory, pageName);
```

Comment:
Could be a different PR but we should consider how not to use the display name for anything but presentation logic as it will cause problems sooner or later.

Reply:
It would be nice to have an additional property in the pages metadata?